examples/enc-search-sample.rs

```diff
@@ -0,0 +1,421 @@
+
+    // Now ask the user for a query string, ask the server for matching records,
+    // and display all matches to the user
+    // start-snippet{executeSearch}
+    let query_str = get_search_query();
+    display_matching_customers(&sdk, &blind_index, &query_str).await?;
+    // end-snippet{executeSearch}
+
+    // Now create a second blind index to use for the email address fields.
+    // start-snippet{createSecondIndex}
+    let encrypted_salt2 = sdk.create_blind_index(&salt_group_id).await?;
+    let blind_index2 = encrypted_salt2.initialize_search(&sdk).await?;
+    // end-snippet{createSecondIndex}
+
+    let mut customer2 = Customer {
+        id: 24,
+        name: "Pokey".to_string(),
+        name_keys: "".to_string(),
+        email: "pokey@gumby.io".to_string(),
+        email_keys: "".to_string(),
+    };
+
+    // start-snippet{useSecondIndex}
+    // Generate the index tokens for the customer name and email address, then encrypt them
+    let name_tokens = blind_index
+        .tokenize_data(&customer2.name, None)?
+        .into_iter()
+        .collect::<Vec<u32>>();
+    let email_tokens = blind_index2
+        .tokenize_data(&customer2.email, None)?
+        .into_iter()
+        .collect::<Vec<u32>>();
+    let enc_name = sdk
+        .document_encrypt_unmanaged(&customer2.name.as_bytes(), &encrypt_opts)
+        .await?;
+    let enc_email = sdk
+        .document_encrypt_unmanaged(&customer2.email.as_bytes(), &encrypt_opts)
+        .await?;
+    // Replace name and email with encoded encrypted versions. Also need to store EDEKs to decrypt both.
+    customer2.name = base64::encode(enc_name.encrypted_data());
+    customer2.name_keys = base64::encode(enc_name.encrypted_deks());
+    customer2.email = base64::encode(enc_email.encrypted_data());
+    customer2.email_keys = base64::encode(enc_email.encrypted_deks());
+    save_customer(customer, &name_tokens, &email_tokens);
+    // end-snippet{useSecondIndex}
+
+    Ok(())
+}
+
+// Load the device context and use it to initialize ironoxide
+async fn initialize_sdk_from_file(device_path: &PathBuf) -> Result<IronOxide> {
+    if device_path.is_file() {
+        let device_context_file = File::open(&device_path)?;
+        let device_context: DeviceContext = serde_json::from_reader(device_context_file)?;
+        println!("Found DeviceContext in \"{}\"", device_path.display());
+        Ok(ironoxide::initialize(&device_context, &Default::default()).await?)
+    } else {
+        Err(AppErr(
+            format!(
+                "Couldn't open file {} containing DeviceContext",
+                device_path.display()
+            )
+            .to_string(),
+        ))
+    }
+}
+
+// Add the program start time to the end of the specified string to
+// form a unique id string, then create a GroupId from it
+// start-snippet{createGroupId}
+fn create_group_id(id_str: &str) -> Result<GroupId> {
+    let stime = &GLOBAL_STATE.read().unwrap().start_time;
```

Comment:
I don't think time should be in GLOBAL_STATE, this is the only place it's used. Just pass it in and store it as state in the main.